### PR TITLE
convert main build.gradle to build.gradle.kts

### PR DESCRIPTION
**Pull request by soloturn**
* convert main build.gradle to build.gradle.kts, and upgrade to gradle-8.10.2
* fix(gestalt-android): remove obsolete gradle `compile` dependencies from publish step

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.10.2-bin.zip
```
Comment (BenjaminAmos) [inline gradle/wrapper/gradle-wrapper.properties]: Upgrading gradle is not an issue but could you please split it out into a separate pull request?

File: README.md [hunk 2]
```diff
@@ -2,3 +2,12 @@
...
 
 For documentations and usage examples, see https://github.com/MovingBlocks/gestalt/wiki
+
+## Development
+To build local, check `Jenkinsfile` on how to. To build `gestalt-android`, do: 
+
+* Download the Android SDK and use the sdkmanager tool to install the API 30 build tools.
+* in the top level directory directory, create a local.properties file by running the
+  command `echo sdk.dir=<SDK_ROOT> > local.properties`, where <SDK_ROOT> is the directory
+  in which you installed the Android SDK.
```
Comment (BenjaminAmos) [inline README.md]:
```suggestion
## Development
To build locally, use `gradlew jar`.
### Android Set-Up (optional)
To build `gestalt-android`, you will need to: 
* Download the Android SDK and use the `sdkmanager` tool to install the API 30 build tools.
* Create a `local.properties` file in the repository root containing:
`sdk.dir=<SDK_ROOT>`,
where <SDK_ROOT> is the directory in which you installed the Android SDK.
```

This might be a better wording?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (soloturn) [reply gradle/wrapper/gradle-wrapper.properties]: done

**Review by BenjaminAmos — DISMISSED** [review]

**Comment by BenjaminAmos** [comment]
The file `gradle/common.gradle` has conficts, so this will need to be rebased before merging.

**Review by BenjaminAmos — APPROVED** [review]
